Classes/DDFileLogger.m

```diff
@@ -1014,10 +1014,9 @@ - (DDLogFileInfo *)lt_currentLogFileInfo {
     }
 
     if (_currentLogFileInfo) {
-        BOOL isMostRecentLogArchived = _currentLogFileInfo.isArchived;
-        BOOL forceArchive = _doNotReuseLogFiles && isMostRecentLogArchived == NO;
-
-        if (forceArchive || [self lt_shouldLogFileBeArchived:_currentLogFileInfo]) {
+        if (_currentLogFileInfo.isArchived) {
+            _currentLogFileInfo = nil;
+        } else if (_doNotReuseLogFiles || [self lt_shouldLogFileBeArchived:_currentLogFileInfo]) {
```

Comment:
These two branches can exist without outer `_currentLogFileInfo != nil`, right?
It will be nice to add `todo` to update method `lt_shouldLogFileBeArchived` to accept nil value. ( and also add `_Nullable` modifier to parameter )